clarify scd table and key for scd table (#5357)
dbt-labs/docs.getdbt.com: website/docs/docs/build/dimensions.md

Currently, there are limitations in supporting SCD's. 
:::

MetricFlow supports joins against dimensions values in a semantic model built on top of a slowly changing dimension (SCD) Type II table. This is useful when you need a particular metric sliced by a group that changes over time, such as the historical trends of sales by a customer's country. 

As their name suggests SCD Type II are groups that change values at a coarser time granularity. This results in a range of valid rows with different dimensions values for a given metric or measure. MetricFlow associates the metric with the first (minimum) available dimensions value within a coarser time window, such as a month. By default, MetricFlow uses the group that is valid at the beginning of the time granularity.

**SCD tables and keys**

> SCD Type II tables have a particular dimension with a start and end date. Use the additional [entity `type`](/docs/build/entities#entity-types) parameters, `natural` key, to join tables If an SCD table has a `natural` key as an [entity `type`](/docs/build/entities#entity-types), you don't usually need a `primary` key. In most instances, SCD tables don't have a logically usable `primary` key because `natural` keys map to multiple rows.

Author: ```suggestion
SCD Type II tables have a particular dimension with a start and end date.  To join tables, use the additional [entity `type`](/docs/build/entities#entity-types) parameter, `natural` key. If an SCD table has a `natural` key as an [entity `type`](/docs/build/entities#entity-types), you don't usually need a `primary` key. In most instances, SCD tables don't have a logically usable `primary` key because `natural` keys map to multiple rows.
```

> SCD Type II tables have a particular dimension with a start and end date.  To join tables, use the additional [entity `type`](/docs/build/entities#entity-types) parameter, `natural` key. If an SCD table has a `natural` key as an [entity `type`](/docs/build/entities#entity-types), you don't usually need a `primary` key. In most instances, SCD tables don't have a logically usable `primary` key because `natural` keys map to multiple rows.

Author: ```suggestion
SCD Type II tables have a particular dimension with a start and end date.  To join tables, use the additional [entity `type`](/docs/build/entities#entity-types) parameter, `natural` key. Using a `natural` key as an [entity `type`](/docs/build/entities#entity-types) means you also don't usually need a `primary` key. In most instances, SCD tables don't have a logically usable `primary` key because `natural` keys map to multiple rows.
```
